src/components/MessagesList/MessagesGroup/Message/MessageButtonsBar/MessageButtonsBar.vue

```diff
@@ -438,13 +450,22 @@ export default {
 				&& this.$store.getters.isActorUser()
 		},
 
+		messageFile() {
+			const firstFileKey = (Object.keys(this.message.messageParameters).find(key => key.startsWith('file')))
+			return this.message.messageParameters[firstFileKey]
+		},
+
 		linkToFile() {
-			if (this.isFileShare) {
-				const firstFileKey = (Object.keys(this.message.messageParameters).find(key => key.startsWith('file')))
-				return this.message.messageParameters?.[firstFileKey]?.link
-			} else {
-				return ''
+			return this.messageFile.link
+		},
+
+		linkToFileDownload() {
+			const uid = getCurrentUser()?.uid
+			if (!uid) {
+				const fileid = this.linkToFile.slice(this.linkToFile.lastIndexOf('/') + 1)
+				return this.messageFile.link + '/download/' + encodeURI(fileid)
 			}
+			return davRemoteURL + '/files/' + encodeURI(uid) + '/' + encodeURI(this.messageFile.path)
```

Comment:
Maybe we'll move it to utils/fileDownload.ts? Feels like a fit to keep reusable logic there